aiogram/api/client/base.py

```diff
@@ -54,14 +62,15 @@ def id(self) -> int:
         await self.session.close()
 
     @asynccontextmanager
-    async def context(self, auto_close: bool = True):
+    async def context(self, auto_close: bool = True) -> AsyncIterator['BaseBot[ContextInstance]']:
         """
         Generate bot context
 
         :param auto_close:
         :return:
         """
-        token = self.set_current(self)
+        # TODO: because set_current expects Bot, not BaseBot — this check fails
+        token = self.set_current(self)  # type: ignore
```

Comment:
this method IRL expects be running only via Bot class, not BaseBot class

If we want an honest typing, we should rewrite BaseBot -> Bot relation

Reply:
We should remove BaseBot class and move it into Bot class